### PR TITLE
Add ability to retry runTest on backend

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

**Review by bjhham — APPROVED** [review]
👍 👍

**Comment by e5l** [comment]
it looks like server is not restarted between repeats, let me check if I can fix it

**Comment by osipxd** [comment]
This failure seems caused by the latest Tomcat update: https://github.com/ktorio/ktor/pull/4470

**Comment by e5l** [comment]
Thanks, will check what has been changed

File: ktor-server/ktor-server-test-base/common/src/io/ktor/server/test/base/BaseTest.kt [hunk 1]
```diff
@@ -4,13 +4,33 @@
...
 expect abstract class BaseTest() {
     open val timeout: Duration
+
+    open fun beforeTest()
+    open fun afterTest()
+
     fun collectUnhandledException(error: Throwable) // TODO: better name?
-    fun runTest(block: suspend CoroutineScope.() -> Unit): TestResult
-    fun runTest(timeout: Duration, block: suspend CoroutineScope.() -> Unit): TestResult
+    fun runTest(timeout: Duration = 60.seconds, block: suspend CoroutineScope.() -> Unit): TestResult
```
Comment (osipxd) [inline ktor-server/ktor-server-test-base/common/src/io/ktor/server/test/base/BaseTest.kt:19]: I'm not sure about such a big default timeout. Successful tests usually run faster.

**Review by osipxd — APPROVED** [review]

File: ktor-server/ktor-server-test-base/common/src/io/ktor/server/test/base/BaseTest.kt (hunk 1 above)
Comment (e5l) [reply ktor-server/ktor-server-test-base/common/src/io/ktor/server/test/base/BaseTest.kt:19]: Could you tell me if you have any value in mind?

Comment (osipxd) [reply ktor-server/ktor-server-test-base/common/src/io/ktor/server/test/base/BaseTest.kt:19]: According to [our Develocity statistics](https://ge.jetbrains.com/scans/tests?search.rootProjectNames=ktor&tests.container=*.server.*&tests.sortField=MEAN_DURATION), 30 seconds should be enough for all tests.

**Pull request merged**